Google.PowerShell/BigQuery/BqDataset.cs

```diff
@@ -242,16 +269,83 @@ public class SetBqDataset : BqCmdlet
         /// dataset in the project specified.
         /// </para>
         /// </summary>
-        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true)]
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.Default)]
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.SetLabel)]
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ClearLabel)]
         public Dataset Dataset { get; set; }
 
+        /// <summary>
+        /// <para type="description">
+        /// Sets the labels in Keys to the values in Values for the target Dataset.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ParameterSetName = ParameterSetNames.SetLabel)]
+        public SwitchParameter SetLabel { get; set; }
```

Comment:
Why don't we just use a HashTable instead of having a SwitchParameter with 2 arrays?